### PR TITLE
Fix @method.route decorator and perform various code improvements

### PR DESCRIPTION
**Pull request by babky-atteq**
Decorator @method.route was not working since Injected import was
missing. README.md is updated with the new information. Documentation of
model_resource is improved along with IdHelper is made abstract with the
 proper metaclasses.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,15 +2,19 @@
 
 ZSL is a Python 2.7 micro-framework utilizing
```
Comment (morihladko) [inline README.md]: It is working in Python 3 so I would strip the (2.7) or add 2.7 and 3.

File: requirements.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ future
 zsl_client
 click>=6.6
 Flask>=0.11
-Flask-Injector==0.8
+Flask-Injector>=0.8
```
Comment (morihladko) [inline requirements.txt]: The version must be set to ==0.8, with 0.9 they went full py3 and broke 2.7 compatibility. The added type annotation to go with the newer python injector DI style.

Comment (babky-atteq) [inline requirements.txt]: Thank you, I wanted to inspect the issue. However, we could make a switch - Py 2.7 = injector ==0.8 and Py 3 gives the latest version. Wouldn't it be possible?

File: README.md (hunk 1 above)
Comment (babky-atteq) [inline README.md]: Fixed.

**Pull request merged**